http-client/src/main/java/io/airlift/http/client/Request.java

```diff
@@ -232,7 +232,7 @@ public Builder setMethod(String method)
 
         public Builder setHeader(String name, String value)
         {
-            this.headers.removeAll(name);
+            this.headers.keys().removeIf(key -> key.equalsIgnoreCase(name));
```

Comment:
good catch!
Does addHeader and addHeaders behave correctly?

Reply:
add are supposed to accumulate

Reply:
Right, it accumulates so not a problem, but there is still a difference and results in names not normalized. But maybe deduplication and normalization is a responsibility of  a different class. 

So it depends if we want:
```        
ListMultimap<String, String> headers = ArrayListMultimap.create();
        headers.put("Key-One", "v1");
        headers.put("Key-One", "v1");
        headers.put("Key-One", "v2");

        headers.put("key-one", "v1");
        
        System.out.println(headers);
```
To result in:
```
{Key-One=[v1, v1, v2], key-one=[v1]}

```
or in:
```
{key-one=[v1, v1, v1, v2]}
```

Reply:
Custom multimap does that:

```
ListMultimap<String, String> headers = Multimaps.newListMultimap(new TreeMap<>(CASE_INSENSITIVE_ORDER), ArrayList::new);
        headers.put("Key-One", "v1");
        headers.put("Key-One", "v1");
        headers.put("Key-One", "v2");
        headers.put("key-one", "v1");
        System.out.println(headers);
```

```
{Key-One=[v1, v1, v2, v1]}
```